<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>Widget</class>
 <widget class="QWidget" name="Widget">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>874</width>
    <height>600</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>HashSearch</string>
  </property>
  <property name="windowIcon">
   <iconset resource="image.qrc">
    <normaloff>:/icon/search.png</normaloff>:/icon/search.png</iconset>
  </property>
  <property name="styleSheet">
   <string notr="true">background-color:#D8D9DA;</string>
  </property>
  <widget class="QLineEdit" name="InputLabel1">
   <property name="geometry">
    <rect>
     <x>30</x>
     <y>40</y>
     <width>591</width>
     <height>31</height>
    </rect>
   </property>
   <property name="font">
    <font>
     <pointsize>12</pointsize>
     <bold>true</bold>
    </font>
   </property>
   <property name="styleSheet">
    <string notr="true">background-color:#61677A;
color:white;
border: 3px solid black;
border-bottom: none; 
border-top-left-radius: 5px;
border-top-right-radius: 5px;</string>
   </property>
   <property name="text">
    <string>输入</string>
   </property>
   <property name="alignment">
    <set>Qt::AlignCenter</set>
   </property>
   <property name="readOnly">
    <bool>true</bool>
   </property>
   <property name="cursorMoveStyle">
    <enum>Qt::LogicalMoveStyle</enum>
   </property>
  </widget>
  <widget class="QPlainTextEdit" name="output">
   <property name="geometry">
    <rect>
     <x>30</x>
     <y>370</y>
     <width>591</width>
     <height>161</height>
    </rect>
   </property>
   <property name="font">
    <font>
     <family>Cascadia Code</family>
     <pointsize>10</pointsize>
    </font>
   </property>
   <property name="styleSheet">
    <string notr="true">background-color:rgb(255, 254, 251);
border: 3px solid black;
border-bottom: none; </string>
   </property>
   <property name="readOnly">
    <bool>true</bool>
   </property>
  </widget>
  <widget class="QLineEdit" name="outputLabel">
   <property name="geometry">
    <rect>
     <x>30</x>
     <y>340</y>
     <width>591</width>
     <height>31</height>
    </rect>
   </property>
   <property name="font">
    <font>
     <pointsize>12</pointsize>
     <bold>true</bold>
    </font>
   </property>
   <property name="styleSheet">
    <string notr="true">background-color:#61677A;
color:white;
border: 3px solid black;
border-bottom: none; 
border-top-left-radius: 5px;
border-top-right-radius: 5px;</string>
   </property>
   <property name="text">
    <string>输出</string>
   </property>
   <property name="alignment">
    <set>Qt::AlignCenter</set>
   </property>
   <property name="readOnly">
    <bool>true</bool>
   </property>
  </widget>
  <widget class="QPlainTextEdit" name="dataInput">
   <property name="geometry">
    <rect>
     <x>30</x>
     <y>70</y>
     <width>591</width>
     <height>201</height>
    </rect>
   </property>
   <property name="font">
    <font>
     <family>Cascadia Code</family>
     <pointsize>10</pointsize>
    </font>
   </property>
   <property name="styleSheet">
    <string notr="true">background-color:rgb(255, 254, 251);
border: 3px solid black;
border-bottom: none; </string>
   </property>
  </widget>
  <widget class="QGroupBox" name="Mapfunction">
   <property name="geometry">
    <rect>
     <x>650</x>
     <y>50</y>
     <width>191</width>
     <height>131</height>
    </rect>
   </property>
   <property name="font">
    <font>
     <family>黑体</family>
     <pointsize>12</pointsize>
    </font>
   </property>
   <property name="styleSheet">
    <string notr="true">color:black; border:3px solid white; border-radius:8px</string>
   </property>
   <property name="title">
    <string>哈希函数类型</string>
   </property>
   <widget class="QRadioButton" name="modTypeHash_RadioButton">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>40</y>
      <width>151</width>
      <height>25</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <family>Microsoft YaHei UI</family>
      <pointsize>11</pointsize>
     </font>
    </property>
    <property name="styleSheet">
     <string notr="true">border:none</string>
    </property>
    <property name="text">
     <string>除留余数法</string>
    </property>
   </widget>
   <widget class="QRadioButton" name="midSquare_RadioButton">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>80</y>
      <width>141</width>
      <height>25</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <family>Microsoft YaHei UI</family>
      <pointsize>11</pointsize>
     </font>
    </property>
    <property name="styleSheet">
     <string notr="true">border:none</string>
    </property>
    <property name="text">
     <string>平方取中法</string>
    </property>
   </widget>
  </widget>
  <widget class="QGroupBox" name="hashCollision">
   <property name="geometry">
    <rect>
     <x>650</x>
     <y>220</y>
     <width>191</width>
     <height>161</height>
    </rect>
   </property>
   <property name="font">
    <font>
     <family>黑体</family>
     <pointsize>12</pointsize>
     <bold>false</bold>
    </font>
   </property>
   <property name="styleSheet">
    <string notr="true">color:black; border:3px solid white; border-radius:8px</string>
   </property>
   <property name="title">
    <string>冲突处理方式</string>
   </property>
   <widget class="QRadioButton" name="randomProbing_RadioButton">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>40</y>
      <width>171</width>
      <height>25</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <family>Microsoft YaHei UI</family>
      <pointsize>11</pointsize>
      <bold>false</bold>
     </font>
    </property>
    <property name="styleSheet">
     <string notr="true">border:none</string>
    </property>
    <property name="text">
     <string>伪随机探测再散列</string>
    </property>
   </widget>
   <widget class="QRadioButton" name="linearProbing_RadioButton">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>80</y>
      <width>171</width>
      <height>25</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <family>Microsoft YaHei UI</family>
      <pointsize>11</pointsize>
      <bold>false</bold>
     </font>
    </property>
    <property name="styleSheet">
     <string notr="true">border:none</string>
    </property>
    <property name="text">
     <string>线性探测再散列</string>
    </property>
   </widget>
   <widget class="QRadioButton" name="quadraticProbing_RadioButton">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>120</y>
      <width>171</width>
      <height>25</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <family>Microsoft YaHei UI</family>
      <pointsize>11</pointsize>
      <bold>false</bold>
     </font>
    </property>
    <property name="styleSheet">
     <string notr="true">border:none</string>
    </property>
    <property name="text">
     <string>二次探测再散列</string>
    </property>
   </widget>
  </widget>
  <widget class="QPushButton" name="hashTableOutput">
   <property name="geometry">
    <rect>
     <x>650</x>
     <y>430</y>
     <width>191</width>
     <height>41</height>
    </rect>
   </property>
   <property name="font">
    <font>
     <pointsize>12</pointsize>
     <bold>true</bold>
    </font>
   </property>
   <property name="styleSheet">
    <string notr="true">QPushButton{
color:white;
background-color: qlineargradient(spread:pad, x1:1, y1:1, x2:1, y2:1, stop:1 #61677A);
border-radius:10px;
}

QPushButton:pressed{
color:white;
background-color: qlineargradient(spread:pad, x1:1, y1:1, x2:1, y2:1, stop:1 rgb(0, 0, 0));
border-radius:10px;
}</string>
   </property>
   <property name="text">
    <string>输出哈希查找表</string>
   </property>
  </widget>
  <widget class="QLineEdit" name="InputLabel2_2">
   <property name="geometry">
    <rect>
     <x>30</x>
     <y>270</y>
     <width>281</width>
     <height>31</height>
    </rect>
   </property>
   <property name="font">
    <font>
     <pointsize>12</pointsize>
     <bold>true</bold>
    </font>
   </property>
   <property name="styleSheet">
    <string notr="true">background-color:#61677A;
color:white;
border: 3px solid black;
border-right: none; 
border-bottom-left-radius: 5px;</string>
   </property>
   <property name="text">
    <string>查找姓名</string>
   </property>
   <property name="alignment">
    <set>Qt::AlignCenter</set>
   </property>
   <property name="readOnly">
    <bool>true</bool>
   </property>
  </widget>
  <widget class="QLineEdit" name="searchNameInput">
   <property name="geometry">
    <rect>
     <x>310</x>
     <y>270</y>
     <width>311</width>
     <height>31</height>
    </rect>
   </property>
   <property name="font">
    <font>
     <family>Cascadia Code</family>
     <pointsize>10</pointsize>
    </font>
   </property>
   <property name="styleSheet">
    <string notr="true">background-color:rgb(255, 254, 251);
border: 3px solid black;
border-bottom-right-radius: 5px;</string>
   </property>
  </widget>
  <widget class="QPushButton" name="SearchIdx">
   <property name="geometry">
    <rect>
     <x>650</x>
     <y>510</y>
     <width>191</width>
     <height>41</height>
    </rect>
   </property>
   <property name="font">
    <font>
     <pointsize>12</pointsize>
     <bold>true</bold>
    </font>
   </property>
   <property name="styleSheet">
    <string notr="true">QPushButton{
color:white;
background-color: qlineargradient(spread:pad, x1:1, y1:1, x2:1, y2:1, stop:1 #61677A);
border-radius:10px;
}

QPushButton:pressed{
color:white;
background-color: qlineargradient(spread:pad, x1:1, y1:1, x2:1, y2:1, stop:1 rgb(0, 0, 0));
border-radius:10px;
}</string>
   </property>
   <property name="text">
    <string>查找映射表下标</string>
   </property>
  </widget>
  <widget class="QLineEdit" name="searchLenOutput">
   <property name="geometry">
    <rect>
     <x>310</x>
     <y>530</y>
     <width>311</width>
     <height>31</height>
    </rect>
   </property>
   <property name="font">
    <font>
     <family>Cascadia Code</family>
     <pointsize>10</pointsize>
    </font>
   </property>
   <property name="styleSheet">
    <string notr="true">background-color:rgb(255, 254, 251);
border: 3px solid black;
border-bottom-right-radius: 5px;</string>
   </property>
   <property name="readOnly">
    <bool>true</bool>
   </property>
  </widget>
  <widget class="QLineEdit" name="InputLabel2">
   <property name="geometry">
    <rect>
     <x>30</x>
     <y>530</y>
     <width>281</width>
     <height>31</height>
    </rect>
   </property>
   <property name="font">
    <font>
     <pointsize>12</pointsize>
     <bold>true</bold>
    </font>
   </property>
   <property name="styleSheet">
    <string notr="true">background-color:#61677A;
color:white;
border: 3px solid black;
border-right: none; 
border-bottom-left-radius: 5px;</string>
   </property>
   <property name="text">
    <string>平均搜索长度</string>
   </property>
   <property name="alignment">
    <set>Qt::AlignCenter</set>
   </property>
   <property name="readOnly">
    <bool>true</bool>
   </property>
  </widget>
 </widget>
 <resources>
  <include location="image.qrc"/>
 </resources>
 <connections/>
</ui>
